### DISCUSSION TITLE
Issue: Page.ColumnA (of type choice) does not work in KQL filter when it has spaces

### DISCUSSION THREAD

**Discussion by mrvdsah** [microsoft-search/pnp-modern-search#3957] [General]
Does not work:

We trying to filter content based on categories. We see that when a category name has a space the query will only pick up the first word in this case Category. So it seems Page.ColumnA content is not filled in correctly by the webparts code. We think it's a bug in the webpart

Column A: "Category A"
Column B: "Category B"
Column C: "Category C"

(  
  (refinablestring03:{Page.ColumnA}) OR
  (refinablestring03:{Page.ColumnB}) OR
  (refinablestring03:{Page.ColumnC}) OR
  (refinablestring04:{Page.ColumnA}) OR
  (refinablestring04:{Page.ColumnB}) OR
  (refinablestring04:{Page.ColumnC}) OR
  (refinablestring05:{Page.ColumnA}) OR
  (refinablestring05:{Page.ColumnB}) OR
  (refinablestring05:{Page.ColumnC})
) AND path:"https://url"

Works:

When filling in hardcode values. The query returns good results.

(  
  (refinablestring03:"Category A") OR
  (refinablestring03:{Page.ColumnB}) OR
  (refinablestring03:{Page.ColumnC}) OR
  (refinablestring04:{Page.ColumnA}) OR
  (refinablestring04:{Page.ColumnB}) OR
  (refinablestring04:{Page.ColumnC}) OR
  (refinablestring05:{Page.ColumnA}) OR
  (refinablestring05:{Page.ColumnB}) OR
  (refinablestring05:{Page.ColumnC})
) AND path:"https://url"

Works 2:

When the categories do not contain any spaces it works. 

Column A: "Category_A"
Column B: "Category_B"
Column C: "Category_C"

(  
  (refinablestring03:{Page.ColumnA}) OR
  (refinablestring03:{Page.ColumnB}) OR
  (refinablestring03:{Page.ColumnC}) OR
  (refinablestring04:{Page.ColumnA}) OR
  (refinablestring04:{Page.ColumnB}) OR
  (refinablestring04:{Page.ColumnC}) OR
  (refinablestring05:{Page.ColumnA}) OR
  (refinablestring05:{Page.ColumnB}) OR
  (refinablestring05:{Page.ColumnC})
) AND path:"https://url"

Version:
4.12.2

**Comment by wobba** [comment]
Take a look at KQL docs https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#grouping-property-restrictions-within-a-kql-query. If you cannot figure it out after reading the docs let us know.
